server.go

```diff
@@ -462,11 +462,85 @@ func (p *sshFxpOpenPacket) respond(svr *Server) responsePacket {
 		osFlags |= os.O_EXCL
 	}
 
-	f, err := os.OpenFile(svr.toLocalPath(p.Path), osFlags, 0o644)
+	name := svr.toLocalPath(p.Path)
+	mode := os.FileMode(0o644)
+
+	var applyAttrs []func(*os.File) error
+
+	// Both `sshFileXferAttrPermissions` and `sshFileXferAttrACmodTime` are set
+	// by e.g. `sftp`. Just in case, we handle all other cases as well.
+	// Only apply for newly created files.
+	var err error
+	useAttrs := true
+	if _, err := os.Stat(name); err == nil {
```

Comment:
Race-condition. The file could come to exist in the time between this test, and the application of these attributes below.

Reply:
Good catch. However, is there any way to solve it? I don't think `OpenFile` gives us this information either. However, `mode` is no-op in `OpenFile` if the file exists. So the only thing I can come up with is that we don't support anything else than mode change, which is a bit disappointing. I've seen time attributes being sent as well.

Reply:
I can only really think of opening with the equivalent of `SSH_FXF_EXCL` , which will fail if the file already exists, which would then take us to a short-circuit path that simply opens without `SSH_FXF_CREAT` and doesn’t apply the settings.

Alternatively, check exactly what what openssh does, and copy that.

Reply:
It seems OpenSSH only uses `mode` in open, so I'll take that route.